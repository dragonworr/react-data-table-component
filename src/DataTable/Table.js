import styled from 'styled-components';

const TableStyle = styled.table`
  height: 100px;
  width: 100%;
  max-width: 100%;
  border-collapse: collapse;
  ${props => props.disabled && 'pointer-events: none'};
  ${props => props.disabled && 'opacity: 0.4'};
`;

export default TableStyle;
