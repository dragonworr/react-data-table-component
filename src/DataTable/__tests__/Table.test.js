import 'jest-styled-components';

import React from 'react';
import { shallow } from 'enzyme';

import Table from '../Table';

test('component <Table /> should render correctly', () => {
  const wrapper = shallow(<Table />);
  expect(wrapper).toMatchSnapshot();
});

test('component <Table disabled={true} /> should render correctly ', () => {
  const wrapper = shallow(<Table disabled />);
  expect(wrapper).toMatchSnapshot();
});
