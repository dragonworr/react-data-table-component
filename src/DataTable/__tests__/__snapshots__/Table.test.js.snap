// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Table /> should render correctly 1`] = `
.c0 {
  height: 100px;
  width: 100%;
  max-width: 100%;
  border-collapse: collapse;
}

<table
  className="c0"
/>
`;

exports[`component <Table disabled={true} /> should render correctly  1`] = `
.c0 {
  height: 100px;
  width: 100%;
  max-width: 100%;
  border-collapse: collapse;
  pointer-events: none;
  opacity: 0.4;
}

<table
  className="c0"
  disabled={true}
/>
`;
